koa-crud/test/integration/controllers/vendor.spec.ts

```diff
@@ -0,0 +1,552 @@
+        ).to.eventually.fulfilled.property('statusCode', 204);
+      });
+    });
+
+    describe('GIVEN correct inputs and TRANSFER type', () => {
+      it('should return a successfull status code', async () => {
+        const main = {
+          body: {
+            name: 'Luis Angelo Belmonte',
+            type: 'TRANSFER',
+          },
+          query: null,
+          params: {
+            id: mock.body.data._id,
+          },
+          ip: null,
+          method: null,
+          path: null,
+          headers: {
+            'Content-Type': null,
+            Referer: null,
+            'User-Agent': null,
+          },
+        };
+
+        await expect(
+          updateVendorController(main),
+        ).to.eventually.fulfilled.property('statusCode', 204);
+      });
+    });
+
+    describe('GIVEN no name', () => {
+      it('should return an error status code', async () => {
+        const main = {
+          body: {
+            name: null,
+            type: 'TRANSFER',
+          },
+          query: null,
+          params: {
+            id: mock.body.data._id,
+          },
+          ip: null,
+          method: null,
+          path: null,
+          headers: {
+            'Content-Type': null,
+            Referer: null,
+            'User-Agent': null,
+          },
+        };
+
+        await expect(
+          updateVendorController(main),
+        ).to.eventually.fulfilled.property('statusCode', 400);
+      });
+    });
+
+    describe('GIVEN no type', () => {
+      it('should return an error status code', async () => {
+        const main = {
+          body: {
+            name: 'Luis Angelo Belmonte',
+            type: null,
+          },
+          query: null,
+          params: {
+            id: mock.body.data._id,
+          },
+          ip: null,
+          method: null,
+          path: null,
+          headers: {
+            'Content-Type': null,
+            Referer: null,
+            'User-Agent': null,
+          },
+        };
+
+        await expect(
+          updateVendorController(main),
+        ).to.eventually.fulfilled.property('statusCode', 400);
+      });
+    });
+
+    describe('GIVEN invalid type', () => {
+      it('should return an error status code', async () => {
+        const main = {
+          body: {
+            name: 'Luis Angelo Belmonte',
+            type: 'qwe',
+          },
+          query: null,
+          params: {
+            id: mock.body.data._id,
+          },
+          ip: null,
+          method: null,
+          path: null,
+          headers: {
+            'Content-Type': null,
+            Referer: null,
+            'User-Agent': null,
+          },
+        };
+
+        await expect(
+          updateVendorController(main),
+        ).to.eventually.fulfilled.property('statusCode', 400);
+      });
+    });
+  });
+
+  describe('Deleting a vendor', () => {
+    after(() => {
+      return Vendor.deleteMany({});
+    });
+
+    let mock;
```

Comment:
Please add a `before` hook here then add this variable as a contextual variable, `this.mock = null`.

Reply:
noted sir